packages/runtime/src/enhancements/default-auth.ts

```diff
@@ -0,0 +1,83 @@
+import { DbClientContract } from '../types';
+import { EnhancementContext, EnhancementOptions } from './create-enhancement';
+import { DefaultPrismaProxyHandler, PrismaProxyActions, makeProxy } from './proxy';
+
+/**
+ * Gets an enhanced Prisma client that supports `@default(auth())` attribute.
+ *
+ * @private
+ */
+export function withDefaultAuth<DbClient extends object>(
+    prisma: DbClient,
+    options: EnhancementOptions,
+    context?: EnhancementContext
+): DbClient {
+    return makeProxy(
+        prisma,
+        options.modelMeta,
+        (_prisma, model) => new DefaultAuthHandler(_prisma as DbClientContract, model, options, context),
+        'defaultAuth'
+    );
+}
+
+class DefaultAuthHandler extends DefaultPrismaProxyHandler {
+    private readonly db: DbClientContract;
+    constructor(
+        prisma: DbClientContract,
+        model: string,
+        private readonly options: EnhancementOptions,
+        private readonly context?: EnhancementContext
+    ) {
+        super(prisma, model);
+        this.db = prisma;
+    }
+
+    // base override
+    protected async preprocessArgs(action: PrismaProxyActions, args: any) {
+        const actionsOfInterest: PrismaProxyActions[] = ['create', 'createMany', 'update', 'updateMany', 'upsert'];
+        if (actionsOfInterest.includes(action)) {
+            await this.preprocessWritePayload(this.model, action as PrismaWriteActionType, args);
+        }
+        return args;
+    }
+
+    private async preprocessWritePayload(model: string, action: PrismaWriteActionType, args: any) {
+        const visitor = new NestedWriteVisitor(this.options.modelMeta, {
+            field: async (field, action, data, context) => {
+                const userContext = this.context?.user;
+                if (!userContext) {
+                    throw new Error(`Invalid user context`);
+                }
+                const fields = this.options.modelMeta.fields[model];
+                const isDefaultAuthField = (fieldInfo: FieldInfo) =>
+                    fieldInfo.attributes?.find((attr) => attr.name === '@default' && attr.args?.[0]?.name === 'auth()');
+                const defaultAuthSelectorFields = Object.fromEntries(
+                    Object.entries(fields)
+                        .filter(([_, fieldInfo]) => isDefaultAuthField(fieldInfo))
+                        .map(([field, fieldInfo]) => [
+                            field,
+                            fieldInfo.attributes?.find((attr) => attr.name === '@default')?.args[0]?.value as
+                                | string
+                                | undefined,
+                        ])
+                );
+                const defaultAuthFields = Object.fromEntries(
+                    Object.entries(defaultAuthSelectorFields).map(([field, selector]) => [
+                        field,
+                        selector ? userContext[selector] : userContext,
```

Comment:
Does it work if there's multi-level selection? Like `@default(auth().profile.image)`.

Reply:
Good point, I don't think so

Reply:
I need your help for this one. I made a `deepGet` helper for the recursive matter (I'm used to use the `get` lodash function, do you already have this kink of helper somewhere in the code ?) but the issue is from the zmodel syntax now. It cannot parse the `auth().a.b` syntax. I tried to update the `zmodel.langium` but without success. I test this case in `auth.test.ts` (last one).

Reply:
Sure. Let me think about how to deal with this.

Reply:
Hi @Azzerty23 , could you help grant me with write permission to your fork of ZenStack? I've made some refactor and would like push it to a separate branch to show the difference, instead of directly committing to this one. Thanks!

Reply:
Sure, I just did it.

Reply:
Thanks! I've just submitted a PR over there, with descriptions on things changed. Please help check if they make sense or there's anything unclear.

Reply:
Wow, that's really nice! The `defaultValueProvider` is so much cleaner. Do you need me to add anything (tests)?
I noticed how you changed the data property directly in the visitor (I had to see it to believe it, ahah). I suppose I can remove my workaround with `newArgs` since it's unnecessary.

Reply:
I think we can keep the `newArgs` there because it's good that the proxy doesn't modify the args that user passes in (instead works on a clone of it).

If it looks good to you, I'll merge the PR to your branch, and let's see if CI passes here 😄. If so, I think the feature is in pretty good shape now, and we can merge it to V2. I'll probably add a set of test cases using Postgres to cover cases not supported by SQLite (e.g., `createMany`). But that'll require some additional setup and a local pg instance. I can do it later.

Thanks for working on this great feature! It'll benefit many people.

Reply:
Okay, I'll let you ignore my last commit (removing `newArgs`) and proceed to merge the PR. Thank you for your prompt and wonderful support!